autopush/router/simple.py

```diff
@@ -188,21 +194,42 @@ def _send_notification(self, uaid, node_id, notification):
                               "version": notification.version,
                               "data": notification.data})
         url = node_id + "/push/" + uaid
-        d = self.ap_settings.agent.request(
+        request = self.ap_settings.agent.request(
             "PUT",
             url.encode("utf8"),
             bodyProducer=FileBodyProducer(StringIO(payload)),
         )
-        d.addCallback(IgnoreBody.ignore)
-        return d
+        # Only time out actual calls, not mocks.
+        if isinstance(request, defer.Deferred):
+            timeout = reactor.callLater(self.ap_settings.router_timeout,
+                                        request.cancel)
```

Comment:
I had figured we'd just drop the connect Timeout, rather than use more code to cancel requests like this (and even though they're cancelled, afaik twisted is still waiting on trying to connect to them).